File: src/test/java/eu/sig/training/ch04/v1/AccountsTest.java
// tag::AccountsTest[]
package eu.sig.training.ch04.v1;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Ignore;
import org.junit.Test;

public class AccountsTest {

    @Test
    public void testIsValidNormalCases() {
        assertTrue(Accounts.isValid("123456789"));
        assertFalse(Accounts.isValid("123456788"));
    }

    // end::AccountsTest[]

    @Ignore("Deliberately fails for illustration purpose")
    // tag::testEmptyString[]
    @Test
    public void testEmptyString() {
        assertFalse(Accounts.isValid(""));
    }
    // end::testEmptyString[]

    @SuppressWarnings("unused")
    @Ignore("Deliberately fails for illustration purpose")
    // tag::showError[]
    @Test
    public void showError() {
        int dummy = 1 / 0;
        // Next line is never executed because the previous one raises an
        // exception.
        // If it were executed, you'll never see the assert message because
        // the test always succeeds.
        assertTrue("You will never see this text.", true);
    }
    // end::showError[]
}

File: src/test/csharp/eu/sig/training/ch04/v1/AccountsTest.cs
// tag::AccountsTest[]
using NUnit.Framework;

namespace eu.sig.training.ch04.v1
{
    [TestFixture]
    public class AccountsTest
    {
        [Test]
        public void TestIsValidNormalCases()
        {
            Assert.IsTrue(Accounts.IsValid("123456789"));
            Assert.IsFalse(Accounts.IsValid("123456788"));
        }

        // end::AccountsTest[]

        // This test is meant to fail.
        [Ignore("Deliberately fails for illustration purpose")]
        // tag::testEmptyString[]
        [Test]
        public void TestEmptyString()
        {
            Assert.IsFalse(Accounts.IsValid(""));
        }
        // end::testEmptyString[]

#pragma warning disable 219
        // This test is meant to fail, that's why we ignore it.
        [Ignore("Deliberately fails for illustration purpose")]
        // tag::showError[]
        [Test]
        public void ShowError()
        {
            int tmp = 0, dummy = 1 / tmp;
            // Next line is never executed because the previous one raises an
            // exception.
            // If it were executed, you'll never see the assert message because
            // the test always succeeds.
            Assert.IsTrue(true);
        }
        // end::showError[]
#pragma warning restore 219
    }
}
